release/models/bgp/openconfig-bgp-policy.yang

```diff
@@ -661,6 +670,297 @@ module openconfig-bgp-policy {
+              container state {
+                  description "State for BGP color extended community
+                  defined by RFC9012";
+                  config false;
+                  uses ext-community-color-config;
+              }
+          }
+        }
+      }
+  }
+
+grouping ext-community-asn-type-config {
+  description "Grouping for asn-type extended community config.";
+  leaf asn {
+      type oc-inet:as-number;
+      description "Autonomous system number.";
+  }
+
+  leaf value {
+      type    uint16;
+      description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-ip-type-config {
+  description "Grouping for ip-type extended community config.";
+  leaf asn {
+    type oc-inet:ip-address;
+    description "IPv4 Address for the community.";
+  }
+
+  leaf value {
+    type    uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-link-bw-config {
+  description "Grouping for link bandwidth extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf bps {
+    type uint64;
+    description "Bandwidth in bits per second.";
+  }
+}
+
+grouping ext-community-route-target-config {
+  description "Grouping for route-target extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-target-ip-config {
+  description "Grouping for route-target-ip extended community
+  config.";
+  leaf ip-address {
+    type oc-inet:ip-address;
+    description "IPv4 address.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-origin-config {
+  description "Grouping for origin extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-origin-ip-config {
+  description "Grouping for route-origin-ip extended community config.";
+  leaf ip-address {
+    type oc-inet:ip-address;
+    description "IPv4 address.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-color-config {
+  description "Grouping for color extended community config.";
+  leaf match-type {
+    type oc-bgp-types:bgp-color-match-type;
+    description "Specifies the match criteria between BGP NH and
+    SR policy endpoint";
+  }
+
+  leaf value {
+    type uint32;
+    description "Assigned value for this community.";
+  }
+}
+
+  grouping ext-community-member-config {
+    description
+    "Configuration data for typed BGP extended communities.";
+
+    leaf index {
+      type uint16;
```

Comment:
What is the full intention of an integer based index here?  Is there reference to any implementation supporting this?

Reply:
The index is intended to be a unique identifier for each record in the list.   It is not intended to identify an order for entries (it's not a sequence id).   (Do we need communities in a community set to be ordered?).   For these reasons, I don't expect index to be represented in an equivalent CLI command or parameter.    

This is one of two options I've thought of so far.  I thought this was preferable to introducing a name based key as community-sets already have a name.  Adding another, nested named for each entry in a community-set seems undesirable.   Since yang requires a list that can be configured to have a key, an unkeyed list isn't an option.

I welcome other ideas on how to model the list of community-set members.   I would like to get rid of the clunky, error-prone regexes we are using to constrain strings to unit16,unit32 values.